prow/github/client.go

```diff
@@ -2355,15 +2378,28 @@ func (c *client) AssignIssue(org, repo string, number int, logins []string) erro
 
 	assigned := make(map[string]bool)
 	var i Issue
-	_, err := c.request(&request{
+	httpStatusCode, githubError, err := c.requestWithGitHubError(&request{
 		method:      http.MethodPost,
 		path:        fmt.Sprintf("/repos/%s/%s/issues/%d/assignees", org, repo, number),
 		requestBody: map[string][]string{"assignees": logins},
-		exitCodes:   []int{201},
+		exitCodes: []int{
+			http.StatusCreated,             //201
+			http.StatusUnprocessableEntity, //422
+		},
 	}, &i)
 	if err != nil {
 		return err
 	}
+	if httpStatusCode == http.StatusUnprocessableEntity {
```

Comment:
I don't think we should be just ignoring that error, we should instead find out how it was possible that we tried to assign the same user twice

Reply:
I agree with you @alvaroaleman 
In fact I had asked for directions on places where I should try to look for the root cause of this issue - [comment 1](https://github.com/kubernetes/test-infra/issues/16145#issuecomment-725786819), [comment 2](https://github.com/kubernetes/test-infra/issues/16145#issuecomment-725872548), [comment 3](https://github.com/kubernetes/test-infra/issues/16145#issuecomment-726739255).
Please note that I'm not familiar with the codebase / workflow to make a judgement on places that could be triggering this type of behavior.

I was thinking that there are two approaches when looking at this type of issues...
1. This is a cloud based solution hence there are chances of calls being sent to your webhook multiple times. Your code should be resilient to such events and work correctly. 
1. You should find out the root cause of why there were 2 calls in the first place.

As I didn't have much knowledge on how to deal with point 2 above, I took this approach to solve point 1 above. I have also left a debug message when we ignore the error incase we wish to still keep looking for the root cause. Is debug level message acceptable?

Please let me know your thoughts.

Reply:
> I was thinking that there are two approaches when looking at this type of issues...

1. This is a cloud based solution hence there are chances of calls being sent to your webhook multiple times. Your code should be resilient to such events and work correctly.
1. You should find out the root cause of why there were 2 calls in the first place.

Regarding 1), that is not expected to happen and if it would happen would cause a lot of other issues as well, like tests always get started twice.

Regarding 2), the place to look would be in the `lgtm` plugin somewhere: https://github.com/kubernetes/test-infra/blob/91b8575a1f1300e0c3b612d466a880d3141c34e3/prow/plugins/lgtm/lgtm.go#L149-L177

I am not sure if the underlying issue qualifies as `good first issue`

Reply:
@alvaroaleman Are we sure about point 1 above not being possible? To rule that out, I've posted a [question in the issue](https://github.com/kubernetes/test-infra/issues/16145#issuecomment-731717641).

Reply:
I meant it as in Github does give this guarantee and Prow would largely stop working properly if that wasn't the case. That being said its possible to configure the same webhook multiple times (for example on the github organization and repository) in which case the event will be sent once per webhook configuration.